WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -388,22 +388,28 @@ private extension ProductsViewController {
         ProductsTopBannerFactory.topBanner(isExpanded: isExpanded,
                                            isInAppFeedbackFeatureEnabled: isInAppFeedbackEnabled,
                                            expandedStateChangeHandler: { [weak self] in
-            self?.tableView.updateHeaderHeight()
+            self?.updateTableHeaderView()
         }, onGiveFeedbackButtonPressed: { [weak self] in
             self?.presentProductsFeedback()
         }, onDismissButtonPressed: { [weak self] in
             self?.dismissProductsBanner()
         }, onCompletion: { [weak self] topBannerView in
             self?.topBannerContainerView.updateSubview(topBannerView)
             self?.topBannerView = topBannerView
-            self?.tableView.updateHeaderHeight()
+            self?.updateTableHeaderView()
         })
     }
 
     func hideTopBannerView() {
         topBannerView?.removeFromSuperview()
         topBannerView = nil
-        tableView.tableHeaderView = nil
+        updateTableHeaderView()
+    }
+
+    /// Updates table header view with the correct spacing / edges depending if `topBannerContainerView` is empty or not.
+    ///
+    func updateTableHeaderView() {
```

Comment:
Do you think `updateTableHeaderViewHeight` would be less ambiguous? 

```suggestion
    func updateTableHeaderViewHeight() {
```

Reply:
I had it like that, but since it does a little bit more than updating just the height I left it as `updateTableHeaderView()` I also had it as `updateTableHeaderViewHeightAndSpacing()` 🤔

Reply:
I guess we can consider changing the spacing as changing the "height" too. 🤣 Anyway, up to you!